src/precice/impl/DataContext.hpp

```diff
@@ -11,18 +11,49 @@ namespace impl {
  * @brief Stores one Data object with related context. If this dataContext is not associated with a mapping,
  * fromData and toData refer to the same data object.
```

Comment:
I guess, we now should add more documentation here. Especially when a DataContext is created and what it stands for exactly.

Reply:
Important points for documentation:

* `toData` and `fromData` are always different, if a mapping exists.
* `toData` and `fromData` don't exist if no mapping exists.
* either `toData` or `fromData` is identical with `participantData` (this depends on whether the mapping is read or write. **todo:** This is something we might want to assert, as well)

Reply:
From my point of view documentation is in good shape now. I have the feeling that as soon as we remove `mapWriteDataFrom` and `mapReadDataTo` (#859) the `DataContext` will become quite simple. I don't think that it's really worth the effort documenting functions that we will remove very soon anyway.